core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -1580,24 +1580,31 @@ public Future<UpdateCenterJob> deploy(boolean dynamicLoad) {
          *      See {@link UpdateCenter#isRestartRequiredForCompletion()}
          * @param correlationId A correlation ID to be set on the job.
          * @param batch if defined, a list of plugins to add to, which will be started later
+         * @param hasEnabledDependents
+         *      If true, this plugin will be enabled if this plugin is disabled.
+         *      If false, this plugin will remain the current status.
          */
         @Restricted(NoExternalUse.class)
-        public Future<UpdateCenterJob> deploy(boolean dynamicLoad, @CheckForNull UUID correlationId, @CheckForNull List<PluginWrapper> batch) {
+        public Future<UpdateCenterJob> deploy(boolean dynamicLoad, @CheckForNull UUID correlationId, @CheckForNull List<PluginWrapper> batch, boolean hasEnabledDependents) {
```

Comment:
@daniel-beck  Actually not all callers pass `false`. [Line 1599](https://github.com/jenkinsci/jenkins/pull/7040/files/0e48fb6466746bb65afb35ca8bd4102c8dffc30a#diff-aeb05543bf9aa3e347768cef830d36f45cd049454384552ee02320cea0f3791bR1599) passes the current status of this plugin (dependent plugin), and it may be `true`.